### DISCUSSION TITLE
[mlp] Working on mlp customization

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#275] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v11

### Description

Hi guys,
we are working on a `mlp` customization for a client.

Is there a way to locally test/debug/troubleshoot mlp or we need to test it in a real environment?

I didn't find help on the "local development" of the tool inside the [_Contributing_ guidelines](https://github.com/mia-platform/mlp/blob/main/CONTRIBUTING.md).

Thanks.

**Comment by davidebianchi** [comment, marked as answer]
Hi! It's more an issue to open on mlp, can you move it to mlp?
What type of extension are you creating?
Anyhow, you could run tests in this way: https://github.com/mia-platform/mlp/blob/main/README.md#testing (tests are running with envtest, so with the real kubernetes services)

If you want a real life manual test, you need to use a real environment (e.g. KinD).

**Reply by frengooo** [reply]
Hi @davidebianchi, thanks, I moved the discussion on the `mlp` repository:
https://github.com/mia-platform/mlp/issues/82

A customization to deploy in a custom runtime, for a PoC.

in the meantime, thanks for the information.
